src/HypreLinearSystem.C

```diff
@@ -1583,7 +1586,7 @@ HypreLinearSystem::HypreLinSysCoeffApplier::sum_into_1DoF(
       for (unsigned k=0; k<numEntities; ++k) {
 	/* binary search subrange rather than a map.find */
 	HypreIntType col = localIds[k];
-	unsigned matIndex=globalNumRows_+1;
+	unsigned matIndex;
```

Comment:
I'm puzzled by this change. Why not initialize matIndex?
Not a big deal, I guess it would be a non-issue if matIndex was a return-value from binarySearchShared, rather than an output argument.

Reply:
If binarySearchShared does not return successfully with a "reasonable value", then there are much bigger problems. I'm reworking this code in another branch. I didn't want to conflate the bug fix which is addressed by the map_shared_.exists() additions with other refactoring.

Reply:
For now, I'm trying to keep the code consistent, i.e. uninitialized. This occurs a couple of other places in HypreLinearSystem and HypreUVWLinearSystem.